src/deployPanel.ts

```
@@ -0,0 +1,206 @@
+        const javascriptHref : string = this.panel.webview.asWebviewUri(
+            vscode.Uri.file(path.join(extensionPath, 'out', 'panels', 'bundles', 'deploy.main.js'))) + '';
+        const cssHref : string = this.panel.webview.asWebviewUri(
+            vscode.Uri.file(path.join(extensionPath, 'out', 'panels', 'deploy.css'))) + '';
+        this.panel.webview.html = htmlFileContents
+            .replace(JavascriptHrefPlaceholder, javascriptHref)
+            .replace(CssHrefPlaceholder, cssHref);
+    }
+
+    public dispose() {
+        this.panel.dispose();
+    }
+
+    private async doDeploy() {
+        try {
+            //
+            // Construct a script that calls Neo.Contract.Create
+            //   -- see: https://docs.neo.org/docs/en-us/reference/scapi/fw/dotnet/neo/Contract/Create.html
+            //
+            // TODO: Find the contracts .abi.json and correctly determine parameters and return types. For now
+            //       we assume that all contracts take two parameters (a string and an array) and return a byte
+            //       array.
+            // TODO: Allow storage usage to be specified in the UI. For now we assume storage is used.
+            // TODO: Allow specification of description, email, etc. in the UI. For now we use blank values.
+            //
+            const sb = neon.default.create.scriptBuilder();
+            const script = sb
+                .emitPush(neon.default.u.str2hexstring('')) // description
+                .emitPush(neon.default.u.str2hexstring('')) // email
+                .emitPush(neon.default.u.str2hexstring('')) // author
+                .emitPush(neon.default.u.str2hexstring('')) // code_version
+                .emitPush(neon.default.u.str2hexstring('')) // name
+                .emitPush(0x01) // storage: {none: 0x00, storage: 0x01, dynamic: 0x02, storage+dynamic:0x03}
+                .emitPush('05') // return type - see https://docs.neo.org/docs/en-us/sc/deploy/Parameter.html
```

This information is in the abi.json file. If the abi.json file is missing, fail deployment rather than assuming return/parameter types